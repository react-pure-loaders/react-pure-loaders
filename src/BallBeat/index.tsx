/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
