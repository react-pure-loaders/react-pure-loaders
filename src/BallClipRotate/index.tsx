/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
