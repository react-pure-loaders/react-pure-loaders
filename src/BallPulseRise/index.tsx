/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
