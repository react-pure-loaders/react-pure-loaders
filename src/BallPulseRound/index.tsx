/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
