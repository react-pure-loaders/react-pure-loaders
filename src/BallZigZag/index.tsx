/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
