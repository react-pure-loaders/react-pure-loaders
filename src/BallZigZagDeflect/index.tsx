/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
