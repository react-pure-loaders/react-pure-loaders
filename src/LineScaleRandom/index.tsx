/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
