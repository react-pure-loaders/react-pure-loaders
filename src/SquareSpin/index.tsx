/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function SquareSpin({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
