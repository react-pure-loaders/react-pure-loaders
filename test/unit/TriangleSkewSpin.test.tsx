import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import TriangleSkewSpin from '../../src/TriangleSkewSpin';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<TriangleSkewSpin>', () => {
    afterEach(cleanup);

    test('TriangleSkewSpin should match snapshot', () => {
        const { container } = render(<TriangleSkewSpin loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('TriangleSkewSpin should have default color', () => {
        const { container } = render(<TriangleSkewSpin loading={true}/>);

        expect(container.firstChild).toHaveStyleRule(
            'border-bottom',
            `20px solid ${PRIMARY_COLOR}`,
            {
                target: '> div',
            },
        );
    });

    test.skip('TriangleSkewSpin should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<TriangleSkewSpin color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule(
            'border-bottom',
            `20px solid ${color}`,
            {
                target: '> div',
            },
        );
    });

    test('TriangleSkewSpin should have no children', () => {
        const { container } = render(<TriangleSkewSpin loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('TriangleSkewSpin should have three children', () => {
        const { container } = render(<TriangleSkewSpin loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(2);
    });
});
