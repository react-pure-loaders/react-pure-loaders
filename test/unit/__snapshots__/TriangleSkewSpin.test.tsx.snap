// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TriangleSkewSpin> TriangleSkewSpin should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    -webkit-transform: perspective(100px) rotateX(0) rotateY(0);
    -moz-transform: perspective(100px) rotateX(0) rotateY(0);
    -ms-transform: perspective(100px) rotateX(0) rotateY(0);
    transform: perspective(100px) rotateX(0) rotateY(0);
  }

  25% {
    -webkit-transform: perspective(100px) rotateX(180deg) rotateY(0);
    -moz-transform: perspective(100px) rotateX(180deg) rotateY(0);
    -ms-transform: perspective(100px) rotateX(180deg) rotateY(0);
    transform: perspective(100px) rotateX(180deg) rotateY(0);
  }

  50% {
    -webkit-transform: perspective(100px) rotateX(180deg) rotateY(180deg);
    -moz-transform: perspective(100px) rotateX(180deg) rotateY(180deg);
    -ms-transform: perspective(100px) rotateX(180deg) rotateY(180deg);
    transform: perspective(100px) rotateX(180deg) rotateY(180deg);
  }

  75% {
    -webkit-transform: perspective(100px) rotateX(0) rotateY(180deg);
    -moz-transform: perspective(100px) rotateX(0) rotateY(180deg);
    -ms-transform: perspective(100px) rotateX(0) rotateY(180deg);
    transform: perspective(100px) rotateX(0) rotateY(180deg);
  }
}

.emotion-0>div {
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 3s 0s cubic-bezier(.09,.57,.49,.9) infinite;
  animation: animation-0 3s 0s cubic-bezier(.09,.57,.49,.9) infinite;
  border-bottom: 20px solid #ffffff;
  border-left: 20px solid transparent;
  border-right: 20px solid transparent;
  height: 0;
  width: 0;
}

<div
  class="emotion-0"
>
  <div />
</div>
`;
